src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,87 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.InvalidInputException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int AGE_GUINNESS_RECORD = 122;
+    private static final int MIN_ID = 0;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_LOGIN_LENGTH = 3;
+    private static final int MAX_LOGIN_LENGTH = 20;
+    private static final int MAX_PASSWORD_LENGTH = 25;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        validateUser(user);
+        return storageDao.add(user);
+    }
+
+    public void validateUser(User user) {
+        if (user == null) {
+            throw new InvalidInputException("User can't be null");
+        }
+
+        String login = user.getLogin();
+        String password = user.getPassword();
+
+        if (login == null) {
+            throw new InvalidInputException("Login can't be null");
+        }
+
+        if (storageDao.get(login) != null) {
```

Comment:
it's a good idea to have ordered checks. For example: first make all checks related to login, after - password, then age.
Also, firstly make all checks about valid email and after testing on existing